sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java

```diff
@@ -0,0 +1,233 @@
+ * }
+ * </pre>
+ *
+ * <h3>Writing Parquet files</h3>
+ *
+ * <p>{@link Write} allows you to write a {@link PCollection} of {@link GenericRecord} into a
+ * Parquet file.
+ *
+ * <p>Like for the {@link Read}, you have to provide the location (path) of the Parquet file and
+ * the Avro schema.
+ *
+ * <p>For example:
+ *
+ * <pre>{@code
+ *  pipeline
+ *    .apply(...) // PCollection<GenericRecord>
+ *    .apply(ParquetIO.write().withPath("/foo/bar").withSchema(schema));
+ * }</pre>
+ */
+public class ParquetIO {
+
+  public static Read read() {
+    return new AutoValue_ParquetIO_Read.Builder().build();
+  }
+
+  public static Write write() {
+    return new AutoValue_ParquetIO_Write.Builder().build();
+  }
+
+  /**
+   * A {@link PTransform} to read from Parquet file.
+   */
+  @AutoValue
+  public abstract static class Read extends PTransform<PBegin, PCollection<GenericRecord>> {
+
+    @Nullable abstract String path();
+    @Nullable abstract Schema schema();
+
+    abstract Builder builder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setPath(String path);
+      abstract Builder setSchema(Schema schema);
+      abstract Read build();
+    }
+
+    /**
+     * Define the location of the Parquet file you want to read.
+     */
+    public Read withPath(String path) {
+      checkArgument(path != null, "ParquetIO.read().withPath(path) called with null path");
+      return builder().setPath(path).build();
+    }
+
+    /**
+     * Define the Avro schema of the record to read from the Parquet file.
+     */
+    public Read withSchema(Schema schema) {
+      checkArgument(schema != null,
+          "ParquetIO.read().withSchema(schema) called with null schema");
+      return builder().setSchema(schema).build();
+    }
+
+    @Override
+    public PCollection<GenericRecord> expand(PBegin input) {
+      return input
+          .apply(Create.of(path()))
+          .apply(ParDo.of(new ReadFn()))
+          .setCoder(AvroCoder.of(schema()));
+    }
+
+    @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
+      builder.add(DisplayData.item("path", path()));
+      builder.add(DisplayData.item("schema", schema().toString()));
+    }
+
+    static class ReadFn extends DoFn<String, GenericRecord> {
+
+      @ProcessElement
+      public void processElement(ProcessContext processContext) throws Exception {
+        Path path = new Path(processContext.element());
+        try (ParquetReader<GenericRecord> reader =
+                 AvroParquetReader.<GenericRecord>builder(path).build()) {
+          GenericRecord read;
+          while ((read = reader.read()) != null) {
+            processContext.output(read);
+          }
+        }
+      }
+
+    }
+
+  }
+
+  /**
+   * {@link PTransform} writing a {@link PCollection} of {@link GenericRecord} into a Parquet file.
+   */
+  @AutoValue
+  public abstract static class Write extends PTransform<PCollection<GenericRecord>, PDone> {
```

Comment:
This should be a `FileIO.Sink` to integrate with `FileIO.write()`.

Reply:
Agree. I'm figuring out a new to use the channel with the `ParquetWriter` to be use in `open()` method.